src/util-buffer.c

```diff
@@ -65,6 +65,11 @@ int MemBufferExpand(MemBuffer **buffer, uint32_t expand_by) {
         return -1;
     }
 
+    /* Adjust expand_by to next multiple of 4k. */
+    if (expand_by % 4096 != 0) {
+        expand_by = expand_by - (expand_by % 4096) + 4096;
```

Comment:
ok too late but why did you choose 4k?

Reply:
Is this good for a low `expand_by` value say 10?
Why don't we let the memory expand up to needed for values < 4096 and then use this logic for the higher values?

Reply:
Victors input is why :)

But in general, its a good idea to expand by some chunk size, instead of just the bare minimum, as its likely to be expanded again. I think 2 common strategies exist:
- some unit of size that makes sense for memory alignment, usually platform specific. 4k I think is common here, so hard to go wrong
- the other common method is doubling I think

But in the end, reallocating can be time consuming, so if asked to expand by 12, may as well expand by something more, as its likely an expansion to 15 may happen in the near future.